src/pages/RFOX/components/Stake/StakeConfirm.tsx

```diff
@@ -247,6 +287,22 @@ export const StakeConfirm: React.FC<StakeConfirmProps & StakeRouteProps> = ({
     )
   }, [stakeAmountUserCurrency, stakingAmountCryptoPrecision, stakingAsset])
 
+  const confirmCopy = useMemo(() => {
+    if (!hasEnoughStakingAssetFeeBalance)
+      return translate('common.insufficientAmountForGas', {
+        assetSymbol: stakingAssetFeeAsset?.symbol,
+        chainSymbol: getChainShortName(stakingAssetFeeAsset?.chainId as KnownChainIds),
+      })
+    if (isApprovalRequired) return translate('common.approve')
+    return translate('RFOX.confirmAndStake')
```

Comment:
actually loving the early returns like this, much more pleasing than `} else if () {` everywhere